NEWS.md

```diff
@@ -70,6 +70,8 @@
 
 14. `fread(x, colClasses="POSIXct")` now also works for columns containing only NA values, [#6208](https://github.com/Rdatatable/data.table/issues/6208). Thanks to @markus-schaffer for the report, and Benjamin Schwendinger for the fix.
 
+15. For `POSIXlt` vectors, `DT[, j := value]` now automatically wraps `j` with `as.POSIXct(values)`, [#1724](https://github.com/Rdatatable/data.table/issues/1724). Thanks to @linzhp for the report, and Benjamin Schwendinger for the fix.
```

Comment:
this is likely to be a breaking change, right? e.g. tests assuming output with POSIXlt columns.

Also, the wording here looks off:

> automatically wraps `j` with `as.POSIXct(values)`

I think there's (1) collision of `j`, the LHS of the example with `j`, the argument to `[.data.table`; and (2) `value` vs. `values`. I'm also not sure we've described what happens precisely. WDYT about

> In `DT[, j := value]`, when `value` is class `POSIXlt`, we automatically coerce it to class `POSIXct` instead.

BTW I looked into the "40 bytes for a single date" thing and I think it's a lot worse? At least on my current system with R 4.1.3:

```r
format(object.size(as.POSIXlt(Sys.time())), 'B')
#  [1] "2344 bytes"
```

This took a shockingly long time to run so saving it here:

```r
DF = data.frame(t=Sys.time() + 0:1000000)
DF$tl = as.POSIXlt(DF$t)
DF$size_t = sapply(seq_len(nrow(DF)), \(i) as.double(object.size(head(DF$t, i))))
DF$size_tl = sapply(seq_len(nrow(DF)), \(i) as.double(object.size(head(DF$tl, i))))
summary(DF$size_tl/DF$size_t)
#    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
#   6.345   6.500   6.500   6.500   6.500   7.797 

head(DF)
#                     t                  tl size_t size_tl
# 1 2024-07-18 17:15:33 2024-07-18 17:15:33    344    2344
# 2 2024-07-18 17:15:34 2024-07-18 17:15:34    352    2360
# 3 2024-07-18 17:15:35 2024-07-18 17:15:35    368    2464
# 4 2024-07-18 17:15:36 2024-07-18 17:15:36    368    2464
# 5 2024-07-18 17:15:37 2024-07-18 17:15:37    384    2640
# 6 2024-07-18 17:15:38 2024-07-18 17:15:38    384    2640
```

So maybe more useful to say "typically takes more than 6 times the storage as an equivalent `POSIXct` column"?